File: https:/github.com/rajesh-kayal-dev/Leetcode.git/882-PeakIndexInAMountainArray/882-PeakIndexInAMountainArray.java
// Last updated: 6/15/2026, 9:59:53 AM
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        //Liner search
        // int i =0;
        // while(arr[i] < arr[i + 1]){
        //     i++;
        // }
        // return i;

        //binary search

        int low = 0, high = arr.length - 1;

        while(low < high){
            // int mid = (low + high)/2;//overflow
            int mid = low + (high - low)/2;

            if(arr[mid] < arr[mid + 1]){
                low = mid + 1;
            }else{
                high = mid;
            }

        }
        return low;
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/882-PeakIndexInAMountainArray/882-PeakIndexInAMountainArray.php
// Last updated: 6/15/2026, 10:00:00 AM
class Solution {

    /**
     * @param Integer[] $arr
     * @return Integer
     */
    function peakIndexInMountainArray($arr) {
        $start = 0;
        $end = count($arr) - 1; // Corrected to use count($arr)
        
        while ($start <= $end) {
            $mid = $start + floor(($end - $start) / 2);
            
            if ($arr[$mid] > $arr[$mid - 1] && $arr[$mid] > $arr[$mid + 1]) {
                return $mid;
            } elseif ($arr[$mid] > $arr[$mid - 1]) {
                $start = $mid + 1;
            } else {
                $end = $mid - 1;
            }
        }
        
        return $mid; // Return the peak index
    }
}
